monitoring/monitors/alchemy/main.js

```diff
@@ -0,0 +1,232 @@
+      tokens: [],
+    },
+    serviceProviderFactory: {
+      name: 'ServiceProviderFactory',
+      address: '0xD17A9bc90c582249e211a4f4b16721e7f65156c8',
+      tokens: [],
+    },
+    ethRewardsManager: {
+      name: 'EthRewardsManager',
+      address: '0x5aa6B99A2B461bA8E97207740f0A689C5C39C3b0',
+      tokens: [],
+    },
+    wormholeClient: {
+      name: 'WormholeClient',
+      address: '0x6E7a1F7339bbB62b23D44797b63e4258d283E095',
+      tokens: [],
+    },
+    trustedNotifierManager: {
+      name: 'TrustedNotifierManager',
+      address: '0x6f08105c8CEef2BC5653640fcdbBE1e7bb519D39',
+      tokens: [],
+    },
+    // ClaimableTokens: {
+    //   address: 'Ewkv3JahEFRKkcJmpoKB7pXbnUHwjAyXiwEo4ZY2rezQ',
+    //   tokens: [],
+    // },
+    // RewardsManager: {
+    //   address: 'DDZDcYdQFEMwcu2Mwo75yGFjJ1mUQyyXLWzhZLEVFcei',
+    //   tokens: [],
+    // },
+    multiSig: {
+      name: 'MultiSig',
+      address: '0xeABCcd75dA6c021c7553dB4A74CACC958812432A',
+      tokens: [],
+    },
+  }
+
+  for (const key in addresses) {
+    const address = addresses[key].address
+
+    // Get token balances
+    const balances = await web3.alchemy.getTokenBalances(address)
+
+    // Remove tokens with zero balance
+    const nonZeroBalances = balances.tokenBalances.filter(token => {
+      return token.tokenBalance !== '0'
+    })
+
+    // Minimize variable churn
+    let name
+    let symbol
+    let decimals
+
+    // Loop through all tokens with non-zero balance
+    for (const token of nonZeroBalances) {
+      // Get balance of token
+      let balance = token.tokenBalance
+
+      if (token.contractAddress === AUDIUS_CONTRACT) {
+        // Skip API call for common request
+        name = 'Audius'
+        decimals = AUDIUS_DECIMALS
+        symbol = 'AUDIO'
+      } else {
+        // Get metadata of non-AUDIO token
+        const metadata = await web3.alchemy.getTokenMetadata(token.contractAddress)
+        name = metadata.name
+        decimals = metadata.decimals
+        symbol = metadata.symbol
+      }
+
+      // Compute token balance in human-readable format
+      balance = balance / Math.pow(10, decimals)
+      balance = balance.toFixed(2)
+
+      // Collection for logs
+      const tokenSummary = {
+        name,
+        balance,
+        symbol,
+      }
+      addresses[key].tokens.push(tokenSummary)
+
+      // export metric for Prometheus
+      METRICS[metricNames.BALANCE].set(
+        { address_name: addresses[key].name },
+        parseFloat(balance),
+      )
+    }
+  }
+
+  console.log(JSON.stringify(addresses))
+}
+const monitorWallets = async () => {
+  try {
+    await scanWallets()
+  } catch (error) {
+    console.log(error)
+    METRICS[metricNames.API_FAILURE].inc()
+  }
+}
+
+const monitorTransfers = async () => {
+  web3.eth.subscribe('logs', {
+    address: AUDIUS_CONTRACT,
+    topics: ['0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef']
```

Comment:
what is this topic?

Reply:
I'm not 100% sure, but every transaction on the token has used that topic to indicate how many tokens have been transferred. Example:

* https://etherscan.io/tx/0xca573914dd6014a8031818412750da5d20c54bca65eeb021eec5ee8d7eaf1568#eventlog

Let me know if I should subscribe to events on other topics as well.

Reply:
i think this topic is the "transfer" event on the token. Lets add a comment to indicate that. i wouldn't have known without looking it up